python/metatensor-operations/metatensor/operations/block_from_array.py

```diff
@@ -48,14 +49,17 @@ def block_from_array(array) -> TensorBlock:
             must have at least two dimensions. Too few provided: {n_dimensions}"
         )
 
+    samples = Labels.range("sample", shape[0])
     components = [
         Labels.range(f"component_{component_index+1}", axis_size)
         for component_index, axis_size in enumerate(shape[1:-1])
     ]
+    properties = Labels.range("property", shape[-1])
 
-    return TensorBlock(
-        values=array,
-        samples=Labels.range("sample", shape[0]),
-        components=components,
-        properties=Labels.range("property", shape[-1]),
-    )
+    device = _dispatch.get_device(array)
+    if device is not None:  # i.e. if array is a torch.Tensor
```

Comment:
I don't think we need to guard on `None` here. `Labels.to` does nothing in metatensor-core, so the code should work fine!